### PR TITLE
doc tokenizers: add a document for TokenNgram

### LINKED ISSUE #892: Document TokenNgram

### PR DESCRIPTION
**Pull request by komainu8**
FIx #892

### CODE REVIEW

**Comment by komainu8** [comment]
@kenhys Could you review this PR, please?

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+.. _token-ngram:
+
+``TokenNgram``
+==============
+
+Summary
+-------
+
+``TokenNgram`` can define its behavior dynamically via its options.
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: It seems that `can change` is better.

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+.. include:: ../../example/reference/tokenizers/token-ngram-uni-gram.log
+.. tokenize --tokenizer 'TokenNgram("n", 1)' --string "Hello World"
+
+Bi-gram:
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-bi-gram.log
+.. tokenize --tokenizer 'TokenNgram("n", 2)' --string "Hello World"
+
+tri-gram:
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: Unify to `Tri- gram`.

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  TokenNgram("unify_symbol", true)
+
+  TokenNgram("unify_digit", true)
+
+Specify multiple options::
+
+  TokenNgram("loose_symbol", true, "loose_blank", true)
+
+``TokenNgram`` also specify multiple options as above.
+We can also specify mingle multiple options except the above example.
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: It is simple to use `We can also specify multiple options at the same time except the above examples.`

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-no-option-with-normalizer.log
+.. tokenize --tokenizer TokenNgram --string "Hello World" --normalizer NormalizerAuto
+
+Advanced usage
+--------------
+
+We can specify multiple options for ``TokenNgram``.
+
+For example, we can deal with orthographic variantsas of phone number by using ``loose_symbol`` and ``loose_blank`` as below.
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: You mean `variants as ...`?

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+This option tells us a location of token of highlight target when we highlight token by ``highlight_html``.
+
+We only use this option when we want to highlight token by ``highlight_html``.
+
+When Groonga tokenize texts that target highlight, always used ``NormalizerAuto`` to normalizer until now.
+Therefore, if we use ``NormalizerNFKC100`` to normalizer, sometimes the position of the highlight shift.
+Because we use different normalizer to normalize token.
+
+This option is its to reduce the shift of the position of the highlight.
+
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: To make it clear, example use case is needed.

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+.. _token-ngram-unify-alphabet:
+
+``unify_alphabet``
+""""""""""""""""""
+
+If we set false, ``TokenNgram`` uses bigram tokenize method for ASCII character.
+
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: it  is kind enough to describe the default value.

Comment (kenhys) [reply doc/source/reference/tokenizers/token_ngram.rst]: you can improve it adding example about ``unify_alphabet``.

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+.. _token-ngram-unify-symbol:
+
+``unify_symbol``
+""""""""""""""""
+
+If we set false, ``TokenNgram`` uses bigram tokenize method for symbols.
+``TokenNgram("unify_symbol", false)`` is same behavior of ``TokenBigramSplitSymbol``.
+
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: it is kind enough to describe the default value.

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+.. _token-ngram-unify-digit:
+
+``unify_digit``
+"""""""""""""""
+
+If we set false, ``TokenNgram`` uses bigram tokenize method for digits.
+
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: it is kind enough to describe the default value.

File: doc/source/reference/tokenizers/token_ngram.rst (hunk 7 above)
Comment (kenhys) [reply doc/source/reference/tokenizers/token_ngram.rst]: you can improve it adding example about ``unify_symbol``.

File: doc/source/reference/tokenizers/token_ngram.rst (hunk 8 above)
Comment (kenhys) [reply doc/source/reference/tokenizers/token_ngram.rst]: you can improve it adding example about `unify_digit`.

**Comment by komainu8** [comment]
@kenhys Thank you for your comments.
I've fixed them.

File: doc/locale/ja/LC_MESSAGES/reference.po [hunk 9]
```diff
@@ -28954,6 +28954,200 @@ msgstr ""
...
+"``TokenBigram``."
+msgstr ""
+"オプションを使わない場合、 ``TokenNgram`` は ``TokenBigram`` と同じ動作になり"
+"ます。"
+
+msgid ""
+"``TokenNgram`` We can also specify multiple options at the same time except "
+"the above examples."
+msgstr "上記の例以外にも複数のオプションを組み合わせて指定できます。"
+
```
Comment (kenhys) [inline doc/locale/ja/LC_MESSAGES/reference.po]: Missing ``TokenNgram`` from translation. Is this expected?

File: doc/source/reference/tokenizers/token_ngram.rst [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+.. load --table Entries
+.. [
+.. {"body": "ア㌕Ａz"}
+.. ]
+.. select Entries   --match_columns body   --query 'グラム'   --output_columns 'highlight_html(body, Terms)'
+
+Because we use different normalizer to normalize token.
+
+This option is its to reduce the shift of the position of the highlight as below.
+
```
Comment (kenhys) [inline doc/source/reference/tokenizers/token_ngram.rst]: `This option is its to reduce` means `This option is used  to reduce ....'?

File: doc/locale/ja/LC_MESSAGES/reference.po (hunk 9 above)
Comment (komainu8) [reply doc/locale/ja/LC_MESSAGES/reference.po]: `TokenNgram` is needless.
I've removed it.

File: doc/source/reference/tokenizers/token_ngram.rst (hunk 10 above)
Comment (komainu8) [reply doc/source/reference/tokenizers/token_ngram.rst]: Yes.
I've fixed it.

**Comment by komainu8** [comment]
@kenhys Thank you for your comments.
I've fixed them.

**Pull request merged**